specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -187,14 +187,36 @@ function Merging({
     [records, handleClose]
   );
 
-  const [resources, setResources] = React.useState<
-    RA<SpecifyResource<AnySchema>>
-  >([]);
-
   const id = useId('merging-dialog');
   const formId = id('form');
   const loading = React.useContext(LoadingContext);
 
+  const [needUpdate, setNeedUpdate] = React.useState(false);
+
+  const rawSpecifyResources = React.useMemo(
+    () => records.map(deserializeResource),
+    [records, needUpdate]
+  );
+
+  const sortedResources = React.useMemo(
+    () =>
+      /*
```

Comment:
this comment might need to be moved or updated.
you should either move it to be right above `(resource) => resource.get('timestampCreated')` and then add a separate comment for `(resource) => resource.get('specifyUser'),` (probably better way)
or modify it to also mention why `(resource) => resource.get('specifyUser'),` is there

sorting by agents that have user first (and if both have user, than user with lowest id first) because agents that have users would presumably have much longer history/more connected records. though you might want to double check that `resource.get('specifyUser')` does not emit warnings if 'specifyUser' fields is not present in the resource (if merging something other than agents in the future)